packages/compiler/server/serverlib.ts

```diff
@@ -465,6 +471,45 @@ export function createServer(host: ServerHost): Server {
     }
   }
 
+  function getNameandSymbolKind(node: Node): { name: string; kind: SymbolKind } | undefined {
+    switch (node.kind) {
+      case SyntaxKind.NamespaceStatement:
+        return { name: node.id.sv, kind: SymbolKind.Namespace };
+      case SyntaxKind.CadlScript:
+        return { name: node.id.sv, kind: SymbolKind.File };
+      case SyntaxKind.EnumStatement:
+        return { name: node.id.sv, kind: SymbolKind.Enum };
+      case SyntaxKind.InterfaceStatement:
+        return { name: node.id.sv, kind: SymbolKind.Interface };
+      case SyntaxKind.OperationStatement:
+        return { name: node.id.sv, kind: SymbolKind.Operator };
+      default:
+        return undefined;
+    }
+  }
+
+  async function getDocumentSymbols(params: DocumentSymbolParams): Promise<SymbolInformation[]> {
+    const file = await compilerHost.readFile(getPath(params.textDocument));
+    const ast = parse(file);
```

Comment:
@nguerrera does this means that we parse the same file over and over again for each one of the hooks? Is that dependnant on caching the syntax tree if unchanged?